faststreams/textio.nim

```diff
@@ -91,6 +92,21 @@ proc writeText*(s: OutputStream, x: CompiledIntTypes) =
   else:
     writeText(s, MatchingUInt(x))
 
+when defined(c):
+  
+  proc writeText*(s: OutputStream, x: float64|float32|float) =
+    ## Write the floating point number to the output stream. It has less overhead
+    ## than `$` because it is directly written to the stream without
+    ## allocating a standard Nim 'string'.
+    ##
+    ## Because this procedure uses stdlib, it is only active when Nim compiles
+    ## with the 'c' flag. Otherwise, float types are passed to the generic writeText
+    ## procedure.
+    var buffer: array[65, char]
+    discard writeFloatToBuffer(buffer, x)
```

Comment:
I don't think you need the `cstring` conversion here. You can use the value returned from `writeFloatToBuffer` as the length of the generated string and that's enough to call `toOpenArray` on the `buffer` directly.

Reply:
Good catch. I'll change this.